plasmapy/dispersion/numerical/hirose_.py

```diff
@@ -0,0 +1,315 @@
+        try:
+            ion = Particle(ion)
+        except TypeError:
+            raise TypeError(
+                f"For argument 'ion' expected type {Particle} but got {type(ion)}."
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError("The particle passed for 'ion' must be an ion or element.")
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.charge_number)
+        except ChargeError:
+            z_mean = 1
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # invalidate T_i argument
+    if "T_i" in kwargs["kwargs"]:
+        raise TypeError(
+            "Got unexpected keyword 'T_i', dispersion relation assumes T_i = 0."
+        )
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must a single value and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if not (k.ndim == 0 or k.ndim == 1):
+        raise ValueError(
+            f"Argument 'k' needs to be a single valued or 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be 0 or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    if theta.ndim not in (0, 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {theta.shape}."
+        )
+
+    # Single k value case
+    if np.isscalar(k.value):
+        k = np.array([k.value]) * u.rad / u.m
+
+    # Calc needed plasma parameters
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=PhysicsWarning)
+        n_e = z_mean * n_i
+        c_s = ion_sound_speed(
+            T_e=T_e,
+            T_i=0 * u.K,
+            ion=ion,
+            n_e=n_e,
+            gamma_e=gamma_e,
+            gamma_i=gamma_i,
+            z_mean=z_mean,
+        ).value
+        v_A = Alfven_speed(B, n_i, ion=ion, z_mean=z_mean).value
+        omega_ci = gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean).value
+        omega_pi = plasma_frequency(n=n_i, particle=ion, z_mean=z_mean).value
```

Comment:
```suggestion
    n_e = z_mean * n_i
    v_A = Alfven_speed(B, n_i, ion=ion, z_mean=z_mean).value
    omega_ci = gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean).value
    omega_pi = plasma_frequency(n=n_i, particle=ion, z_mean=z_mean).value
    with warnings.catch_warnings():
        warnings.simplefilter("ignore", category=PhysicsWarning)
        c_s = ion_sound_speed(
            T_e=T_e,
            T_i=0 * u.K,
            ion=ion,
            n_e=n_e,
            gamma_e=gamma_e,
            gamma_i=gamma_i,
            z_mean=z_mean,
        ).value
```

Since `ion_sound_speed` is the only function we expect this warning from, we ought to restrict the catch to that function and let the other happen because we did not "expect" them to happen.